core/src/readwriter.rs

```diff
@@ -39,6 +39,732 @@ impl ToSchema for &str {
+}
+
+impl OrcOptions {
+    pub fn new() -> Self {
+        OrcOptions {
+            merge_schema: None,
+            path_glob_filter: None,
+            recursive_file_lookup: None,
+            modified_before: None,
+            modified_after: None,
+        }
+    }
+
+    pub fn merge_schema(mut self, value: bool) -> Self {
+        self.merge_schema = Some(value);
+        self
+    }
+
+    pub fn path_glob_filter(mut self, value: &str) -> Self {
+        self.path_glob_filter = Some(value.to_string());
+        self
+    }
+
+    pub fn recursive_file_lookup(mut self, value: bool) -> Self {
+        self.recursive_file_lookup = Some(value);
+        self
+    }
+
+    pub fn modified_before(mut self, value: &str) -> Self {
+        self.modified_before = Some(value.to_string());
+        self
+    }
+
+    pub fn modified_after(mut self, value: &str) -> Self {
+        self.modified_after = Some(value.to_string());
+        self
+    }
+}
+
+impl ConfigOpts for OrcOptions {
+    fn to_options(&self) -> HashMap<String, String> {
+        let mut options: HashMap<String, String> = HashMap::new();
+
+        if let Some(merge_schema) = self.merge_schema {
+            options.insert("merge_schema".to_string(), merge_schema.to_string());
+        }
+
+        if let Some(path_glob_filter) = &self.path_glob_filter {
+            options.insert("path_glob_filter".to_string(), path_glob_filter.to_string());
+        }
+
+        if let Some(recursive_file_lookup) = self.recursive_file_lookup {
+            options.insert(
+                "recursive_file_lookup".to_string(),
+                recursive_file_lookup.to_string(),
+            );
+        }
+
+        if let Some(modified_before) = &self.modified_before {
+            options.insert("modified_before".to_string(), modified_before.to_string());
+        }
+
+        if let Some(modified_after) = &self.modified_after {
+            options.insert("modified_after".to_string(), modified_after.to_string());
+        }
+
+        options
+    }
+}
+
+/// A struct that represents options for configuring Parquet file parsing.
+///
+/// # Options
+///
+/// - `merge_schema`: Merge schemas from all Parquet files.
+/// - `path_glob_filter`: A glob pattern to filter files by their path.
+/// - `recursive_file_lookup`: Enable recursive search for files in directories.
+/// - `modified_before`: Only include files modified before a given timestamp.
+/// - `modified_after`: Only include files modified after a given timestamp.
+/// - `datetime_rebase_mode`: Controls how Spark handles rebase of datetime fields.
+/// - `int96_rebase_mode`: Controls how Spark handles rebase of INT96 fields.
+///
+/// # Example
+/// ```
+/// let options = ParquetOptions::new()
+///     .merge_schema(true)
+///     .path_glob_filter("*.parquet".to_string())
+///     .recursive_file_lookup(true);
+///
+/// let df = spark.read().parquet("/path/to/parquet", options)?;
+/// ```
+#[derive(Debug)]
+pub struct ParquetOptions {
```

Comment:
can you also add the `compression` option?

Reply:
Just added the `compression` for the `Option` structs.